src/sentry/api/endpoints/organization_spans_trace.py

```diff
@@ -12,20 +13,23 @@
 from sentry.api.paginator import GenericOffsetPaginator
 from sentry.api.utils import handle_query_errors, update_snuba_params_with_timestamp
 from sentry.models.organization import Organization
-from sentry.search.events.builder.spans_indexed import SpansIndexedQueryBuilder
 from sentry.search.events.types import SnubaParams
-from sentry.snuba.dataset import Dataset
 from sentry.snuba.referrer import Referrer
-from sentry.utils.validators import INVALID_ID_DETAILS, is_event_id
+from sentry.snuba.spans_rpc import RPCSpan, run_trace_query
 
 
-class SnubaTrace(TypedDict):
-    span_op: str
-    span_description: str
-    id: str
+class SerializedEvent(TypedDict):
```

Comment:
@wmak thinking out loud: If we want the FE to MOSTLY work out of the box we should adhere to the [RawSpanType](https://github.com/getsentry/sentry/blob/master/static/app/components/events/interfaces/spans/types.tsx#L32-L54) i.e. changing `event_id` to `span_id`, `end_timestamp` to `timestamp` for now, removing `transaction`(or maybe adding an event_type='span'. Today we check if event has `transaction` to label it as a `transaction` in the tree). With these changes we wouldn't have to worry about adding a new type of span for all the guards in place and we won't have to do too many conditionals. We would just have to use `is_transaction` for the collapsing logic. 

Also, can we add op to the response. We don't need duration.

Reply:
@Abdkhan14 
Actually thinking about it, i rather update the field names so they're clearer than to match the existing response.
ie.
`event_id` -> There will be ids that aren't just spans
`end_timestamp` -> because you need `end` and `start` and `timestamp` is ambiguous
`transaction` -> 🤔 happy to remove this, its just the txn name, but it exists in the old response too?

Can you actually get this working out of the box given that the response now includes spans along with txn?